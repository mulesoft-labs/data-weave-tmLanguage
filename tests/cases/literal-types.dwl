type CardinalDirection = "North" | "East" | "South" | "West"

type OneToThree = 1 | 2 | 3

type QubitBool = true | false | "Superposition"

fun foo(a: 123, b: "Hola" | "Chau"): true = true
---
{}