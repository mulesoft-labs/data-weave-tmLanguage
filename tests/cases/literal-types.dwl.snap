>type CardinalDirection = "North" | "East" | "South" | "West"
#^^^^ source.data-weave meta.directive.type.dw storage.type.dw
#    ^ source.data-weave meta.directive.type.dw
#     ^^^^^^^^^^^^^^^^^ source.data-weave meta.directive.type.dw entity.name.type.dw
#                      ^ source.data-weave meta.directive.type.dw
#                       ^ source.data-weave meta.directive.type.dw keyword.other.dw
#                        ^ source.data-weave meta.directive.type.dw
#                         ^ source.data-weave meta.directive.type.dw string.quoted.double.dw punctuation.definition.string.begin.dw
#                          ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                           ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                            ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                             ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                              ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                               ^ source.data-weave meta.directive.type.dw string.quoted.double.dw punctuation.definition.string.end.dw
#                                ^ source.data-weave meta.directive.type.dw
#                                 ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#                                  ^ source.data-weave meta.directive.type.dw
#                                   ^ source.data-weave meta.directive.type.dw string.quoted.double.dw punctuation.definition.string.begin.dw
#                                    ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                     ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                      ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                       ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                        ^ source.data-weave meta.directive.type.dw string.quoted.double.dw punctuation.definition.string.end.dw
#                                         ^ source.data-weave meta.directive.type.dw
#                                          ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#                                           ^ source.data-weave meta.directive.type.dw
#                                            ^ source.data-weave meta.directive.type.dw string.quoted.double.dw punctuation.definition.string.begin.dw
#                                             ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                              ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                               ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                                ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                                 ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                                  ^ source.data-weave meta.directive.type.dw string.quoted.double.dw punctuation.definition.string.end.dw
#                                                   ^ source.data-weave meta.directive.type.dw
#                                                    ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#                                                     ^ source.data-weave meta.directive.type.dw
#                                                      ^ source.data-weave meta.directive.type.dw string.quoted.double.dw punctuation.definition.string.begin.dw
#                                                       ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                                        ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                                         ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                                          ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                                           ^ source.data-weave meta.directive.type.dw string.quoted.double.dw punctuation.definition.string.end.dw
>
>type OneToThree = 1 | 2 | 3
#^^^^ source.data-weave meta.directive.type.dw storage.type.dw
#    ^ source.data-weave meta.directive.type.dw
#     ^^^^^^^^^^ source.data-weave meta.directive.type.dw entity.name.type.dw
#               ^ source.data-weave meta.directive.type.dw
#                ^ source.data-weave meta.directive.type.dw keyword.other.dw
#                 ^ source.data-weave meta.directive.type.dw
#                  ^ source.data-weave meta.directive.type.dw constant.numeric.dw
#                   ^ source.data-weave meta.directive.type.dw
#                    ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#                     ^ source.data-weave meta.directive.type.dw
#                      ^ source.data-weave meta.directive.type.dw constant.numeric.dw
#                       ^ source.data-weave meta.directive.type.dw
#                        ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#                         ^ source.data-weave meta.directive.type.dw
#                          ^ source.data-weave meta.directive.type.dw constant.numeric.dw
>
>type QubitBool = true | false | "Superposition"
#^^^^ source.data-weave meta.directive.type.dw storage.type.dw
#    ^ source.data-weave meta.directive.type.dw
#     ^^^^^^^^^ source.data-weave meta.directive.type.dw entity.name.type.dw
#              ^ source.data-weave meta.directive.type.dw
#               ^ source.data-weave meta.directive.type.dw keyword.other.dw
#                ^ source.data-weave meta.directive.type.dw
#                 ^^^^ source.data-weave meta.directive.type.dw constant.language.dw
#                     ^ source.data-weave meta.directive.type.dw
#                      ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#                       ^ source.data-weave meta.directive.type.dw
#                        ^^^^^ source.data-weave meta.directive.type.dw constant.language.dw
#                             ^ source.data-weave meta.directive.type.dw
#                              ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#                               ^ source.data-weave meta.directive.type.dw
#                                ^ source.data-weave meta.directive.type.dw string.quoted.double.dw punctuation.definition.string.begin.dw
#                                 ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                  ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                   ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                    ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                     ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                      ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                       ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                        ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                         ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                          ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                           ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                            ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                             ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#                                              ^ source.data-weave meta.directive.type.dw string.quoted.double.dw punctuation.definition.string.end.dw
>
>fun foo(a: 123, b: "Hola" | "Chau"): true = true
#^^^ source.data-weave meta.directive.fun.dw storage.type.dw
#   ^ source.data-weave meta.directive.fun.dw
#    ^^^ source.data-weave meta.directive.fun.dw entity.name.function.dw
#       ^ source.data-weave meta.directive.fun.dw
#        ^ source.data-weave meta.directive.fun.dw variable.parameter.dw
#         ^ source.data-weave meta.directive.fun.dw keyword.operator.declaration.dw
#          ^ source.data-weave meta.directive.fun.dw
#           ^^^ source.data-weave meta.directive.fun.dw constant.numeric.dw
#              ^^ source.data-weave meta.directive.fun.dw
#                ^ source.data-weave meta.directive.fun.dw variable.parameter.dw
#                 ^ source.data-weave meta.directive.fun.dw keyword.operator.declaration.dw
#                  ^ source.data-weave meta.directive.fun.dw
#                   ^ source.data-weave meta.directive.fun.dw string.quoted.double.dw punctuation.definition.string.begin.dw
#                    ^ source.data-weave meta.directive.fun.dw string.quoted.double.dw
#                     ^ source.data-weave meta.directive.fun.dw string.quoted.double.dw
#                      ^ source.data-weave meta.directive.fun.dw string.quoted.double.dw
#                       ^ source.data-weave meta.directive.fun.dw string.quoted.double.dw
#                        ^ source.data-weave meta.directive.fun.dw string.quoted.double.dw punctuation.definition.string.end.dw
#                         ^ source.data-weave meta.directive.fun.dw
#                          ^ source.data-weave meta.directive.fun.dw keyword.operator.declaration.dw
#                           ^ source.data-weave meta.directive.fun.dw
#                            ^ source.data-weave meta.directive.fun.dw string.quoted.double.dw punctuation.definition.string.begin.dw
#                             ^ source.data-weave meta.directive.fun.dw string.quoted.double.dw
#                              ^ source.data-weave meta.directive.fun.dw string.quoted.double.dw
#                               ^ source.data-weave meta.directive.fun.dw string.quoted.double.dw
#                                ^ source.data-weave meta.directive.fun.dw string.quoted.double.dw
#                                 ^ source.data-weave meta.directive.fun.dw string.quoted.double.dw punctuation.definition.string.end.dw
#                                  ^ source.data-weave meta.directive.fun.dw
#                                   ^ source.data-weave meta.directive.fun.dw keyword.operator.declaration.dw
#                                    ^ source.data-weave meta.directive.fun.dw
#                                     ^^^^ source.data-weave meta.directive.fun.dw constant.language.dw
#                                         ^ source.data-weave meta.directive.fun.dw
#                                          ^ source.data-weave meta.directive.fun.dw keyword.operator.assignment.dw
#                                           ^ source.data-weave
#                                            ^^^^ source.data-weave constant.language.dw
>---
#^^^ source.data-weave keyword.operator.body-marker.dw
>{}
#^ source.data-weave meta.objectliteral.dw punctuation.definition.block.dw
# ^ source.data-weave meta.objectliteral.dw punctuation.definition.block.dw